pkg/controller/jobs/statefulset/statefulset_webhook.go

```diff
@@ -142,21 +142,23 @@ func (wh *Webhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Ob
 		groupNameLabelPath,
 	)...)
 
-	oldReplicas := ptr.Deref(oldStatefulSet.Spec.Replicas, 1)
-	newReplicas := ptr.Deref(newStatefulSet.Spec.Replicas, 1)
-
-	// Allow only scale down to zero and scale up from zero.
-	// TODO(#3279): Support custom resizes later
-	if newReplicas != 0 && oldReplicas != 0 {
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(
-			newStatefulSet.Spec.Replicas,
-			oldStatefulSet.Spec.Replicas,
-			replicasPath,
-		)...)
-	}
+	if newQueueName != "" {
```

Comment:
This covers 90% of cases. However, ideally, even if the StatefulSet has queue-name, but the owner is already managed by Kueue, then we don't do the validation. 

It is more of a corner case, but this could matter for the LWS integration or AppWrapper. 

Since this is a corner case I'm on a fence. I'm ok to merge as is, but maybe try to add the check @mbobrovskyi ?

Reply:
Maybe we introduce a helper function `isManagedByKueue`, and do both checks?
```
if parent managed by Kueue then 
 return false
else if hasQueueName
  return true
 return false
```